test-file fails on non-english system

On my french system, `test/file.jl` fails because the error message is in French and not in English

`Aucun fichier ou dossier de ce type` means `No such file or directory`

```bash
$ make test-file
(...)
Test Summary: | Pass  Fail  Total
  Overall     | 1034     1   1035
    file      | 1034     1   1035
    FAILURE

The global RNG seed was 0xdd93b9c7cbba68718d166f5e04dc0622.

Error in testset file:
Test Failed at /home/marc/workspace/julia/test/file.jl:525
  Expression: sprint(showerror, e) == "SystemError: opening file \"this file is not expected to exist\": No such file or directory"
   Evaluated: "SystemError: opening file \"this file is not expected to exist\": Aucun fichier ou dossier de ce type" == "SystemError: opening file \"this file is not expected to exist\": No such file or directory"
ERROR: LoadError: Test run finished with errors
in expression starting at /home/marc/workspace/julia/test/runtests.jl:81
Makefile:25: recipe for target 'file' failed
make[1]: *** [file] Error 1
Makefile:594: recipe for target 'test-file' failed
make: *** [test-file] Error 2
```

```julia
julia> versioninfo()
Julia Version 1.6.0-DEV.1046
Commit 55aeb2ff01* (2020-09-25 08:02 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-10.0.1 (ORCJIT, skylake)

```

--- a/test/file.jl
+++ b/test/file.jl
@@ -604,7 +604,8 @@ close(s)
         false
     catch e
         isa(e, SystemError) || rethrow()
-        @test sprint(showerror, e) == "SystemError: opening file \"this file is not expected to exist\": No such file or directory"
+        @test e.errnum == 2
+        @test startswith(sprint(showerror, e), "SystemError: opening file \"this file is not expected to exist\"")
         true
     end
 end

